cdk/lib/sender-stack.ts

```diff
@@ -0,0 +1,145 @@
+import { GuAutoScalingGroup } from '@guardian/cdk/lib/constructs/autoscaling';
+import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
+import { AppIdentity, GuStack } from '@guardian/cdk/lib/constructs/core';
+import { GuVpc, SubnetType } from '@guardian/cdk/lib/constructs/ec2';
+import {
+	GuAllowPolicy,
+	GuInstanceRole,
+} from '@guardian/cdk/lib/constructs/iam';
+import type { GuAsgCapacity } from '@guardian/cdk/lib/types';
+import type { App } from 'aws-cdk-lib';
+import { CfnOutput, Duration } from 'aws-cdk-lib';
+import { HealthCheck, ScalingEvents } from 'aws-cdk-lib/aws-autoscaling';
+import { InstanceClass, InstanceSize, InstanceType } from 'aws-cdk-lib/aws-ec2';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Topic } from 'aws-cdk-lib/aws-sns';
+import { Queue } from 'aws-cdk-lib/aws-sqs';
+import {
+	ParameterDataType,
+	ParameterTier,
+	StringParameter,
+} from 'aws-cdk-lib/aws-ssm';
+
+interface SenderStackProps extends GuStackProps {
+	appName: string;
+	asgCapacity: GuAsgCapacity;
+	instanceType: InstanceType;
+	notificationSnsTopic: string;
+	cleanerQueueArn: string;
+}
+
+export class SenderWorkerStack extends GuStack {
+	constructor(scope: App, id: string, props: SenderStackProps) {
+		super(scope, id, props);
+
+		const sqsMessageVisibilityTimeout = Duration.seconds(100);
+		const sqsMessageRetentionPeriod = Duration.hours(1);
+		const sqsMessageRetryCount = 5;
+		const instanceType = InstanceType.of(InstanceClass.T4G, InstanceSize.SMALL);
+		const targetUtilizationPercent = 40;
+
+		const vpc = GuVpc.fromIdParameter(
+			this,
+			AppIdentity.suffixText({ app: props.appName }, 'VPC'),
+		);
+		const distributionRole = new GuInstanceRole(this, {
+			app: props.appName,
+			additionalPolicies: [
```

Comment:
do we get permission to emit logs (to cloudwatch?) by default?

Reply:
The GuCDK class grants the permission to write to the aggregated log stream (`mobile-log-aggregation-${stage}` in mobile account).  Usually in EC2, programs write log to a log file, and an external program `aws-kinesis-agent` is used to move the logs to the kinesis stream.